src/sources/gdal.jl

```diff
@@ -391,3 +391,66 @@ build_geotransform(lat, lon) = begin
     gt[GDAL_NS_RES] = step(lat)
     return gt
 end
+
+function GeoArray(dataset::AG.Dataset, key = nothing;
+                  usercrs = nothing,
+                  dims = dims(AG.RasterDataset(dataset), usercrs),
+                  refdims = (),
+                  name = "",
+                  metadata = metadata(AG.RasterDataset(dataset)),
+                  missingval = missingval(AG.RasterDataset(dataset)))
+    GeoArray(AG.read(dataset), dims, refdims, name, metadata, missingval)
+end
+
+function unsafe_ArchGDALdataset(A::GeoArray)
+    width = size(A, Lon)
+    height = size(A, Lat)
+    nbands = size(A, Band)
```

Comment:
We can't actually be sure that a `GeoArray` has any of these dimensions - especially `Band` but the others as well as it could have been sliced. You can use e.g. `if hasdim(A, Lon)` to check if `A` has a `Lat` dim.

You can make a Dataset from 2d without a Band - check the `write` methods.

The GeoArray may also have other dimensions... I was wondering if it would be cool to just loop over the time/vertical etc dimension somewhere and slice it - then convert each slice with GDAL and cat them together afterwards. But don't worry about that now :)

Reply:
Ahh I see -- is there a way to identify the X dimension, the Y dimension, and the number of bands without relying on dim names?

Reply:
They should be Lat/Lon/Band for GeoArray - but a GeoArray might be just a vector after indexing or something and not have them all anymore. ANd other formats like netcdf wont have `Band` - they might just be 2d lat/lon arrays. 

But maybe don't worry about this for now I'll fix it up later - I'm distracting you from the main use case with the edge cases.